src/whispercpp/context.cc

```diff
@@ -229,7 +229,8 @@ std::string Context::sys_info() {
 // Uses the specified decoding strategy to obtain the text. This is
 // usually the only function you need to call as an end user.
 int Context::full(Params params, std::vector<float> data) {
-  int ret = whisper_full(ctx, params.wfp, data.data(), data.size());
+  Params copy = params.copy_for_full(*this);
```

Comment:
why do we need to copy the params here? I guess because the Context class is mutable now?